subsys/bluetooth/mesh/brg_cfg.c

```diff
@@ -0,0 +1,287 @@
+{
+	int err;
+
+	if (len_rd == 0) {
+		brg_enabled = 0;
+		LOG_DBG("Cleared brgen");
+		return 0;
+	}
+
+	err = bt_mesh_settings_set(read_cb, cb_arg, &brg_enabled, sizeof(brg_enabled));
+	if (err) {
+		LOG_ERR("Failed to set brgen");
+		return err;
+	}
+
+	LOG_DBG("Restored sbren");
+
+	return 0;
+}
+
+/* Define a setting for storing enable state */
+BT_MESH_SETTINGS_DEFINE(brgen, "brgen", brgen_set);
+
+/* Set function for initializing bridging table rows from brg_tbl slist */
+static int brg_tbl_set(const char *name, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	if (len_rd == 0) {
+		memset(brg_tbl, 0, sizeof(brg_tbl));
+		LOG_DBG("Cleared brg_tbl");
+		return 0;
+	}
+
+	int err = bt_mesh_settings_set(read_cb, cb_arg, &brg_tbl, sizeof(brg_tbl));
+
+	if (err) {
+		LOG_ERR("Failed to set brg_tbl");
+		return err;
+	}
+
+	LOG_DBG("Restored brg_tbl");
+
+	return 0;
+}
+
+/* Define a setting for storing briging table rows */
+BT_MESH_SETTINGS_DEFINE(brgbt, "brgbt", brg_tbl_set);
+
+bool brg_cfg_enable_get(void)
+{
+	return brg_enabled;
+}
+
+int brg_cfg_enable_set(bool enable)
+{
+	if (brg_enabled == enable) {
+		return 0;
+	}
+
+	brg_enabled = enable;
+
+	bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_SBR_PENDING);
+
+	return 0;
+}
+
+void bt_mesh_brg_en_pending_store(void)
+{
+#if CONFIG_BT_MESH_BRG_CFG_SRV
+	char *path = "bt/mesh/brgen";
+	int err;
+
+	err = settings_save_one(path, &brg_enabled, sizeof(brg_enabled));
+
+	if (err) {
+		LOG_ERR("Failed to store %s value", path);
+	}
+#endif
+}
+
+void bt_mesh_brg_tbl_pending_store(void)
+{
+#if CONFIG_BT_MESH_BRG_CFG_SRV
+	char *path = "bt/mesh/brgbt";
+	char path_bt[20];
+	int err;
+
+	err = settings_save_one(path, &brg_tbl, brg_tbl_row_cnt * sizeof(brg_tbl[0]));
+
+	if (err) {
+		LOG_ERR("Failed to store %s value", path_bt);
+	}
+#endif
+}
+
+/* Remove the entry from the bridging table that corresponds with the netkey index
+ * of the removed subnet.
+ */
+static void sbr_netkey_removed_evt(struct bt_mesh_subnet *sub, enum bt_mesh_key_evt evt)
+{
+	for (int i = 0; i < CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX; i++) {
+		if (brg_tbl[i].direction && (
+		    brg_tbl[i].netidx1 == sub->net_idx ||
+		    brg_tbl[i].netidx2 == sub->net_idx)) {
+			memset(&brg_tbl[i], 0, sizeof(brg_tbl[i]));
+			brg_tbl_compact();
+		}
+	}
+
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+	bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_SBR_PENDING);
+#endif
+}
+
+/* Add event hook for key deletion event */
+BT_MESH_SUBNET_CB_DEFINE(sbr) = {
+	.evt_handler = sbr_netkey_removed_evt,
+};
+
+int brg_cfg_init(void)
+{
+	memset(brg_tbl, 0, sizeof(brg_tbl));
+	brg_tbl_row_cnt = 0;
+	return 0;
+}
+
+int brg_cfg_tbl_reset(void)
+{
+	brg_cfg_init();
+
+	int err = settings_delete("bt/mesh/brgen");
+
+	if (err) {
+		return err;
+	}
+	err = settings_delete("bt/mesh/brgbt");
+
+	return err;
+}
+
+const struct brg_tbl_row *brg_cfg_tbl_get(void)
```

Comment:
This should be added to the h-file.
And should be change this to contain a const pointer as input and the number of elements as return value.
`uint32_t brg_tbl_get(const struct brg_tbl_row *row)`
This way we know how may elements we can iterate over.